### DISCUSSION TITLE
Scala Support

### DISCUSSION THREAD

**Discussion by rockpunk** [facebook/buck2#727] [Q&A]
Hi, I'm a bit confused as to Buck2's support for scala rules. I notice that there's declarations for `scala_test` and `scala_library` in the [buck2-prelude](https://github.com/facebook/buck2-prelude/blob/main/decls/scala_rules.bzl), but the implementation is missing.  I thought it might be as easy borrowing directly from the bazel community's [rules_scala](https://github.com/bazelbuild/rules_scala) implementation, but I'm at a loss for how to start. Can anyone in the buck2 community point me to an example of a working minimal scala setup?  Thank you!

**Comment by ndmitchell** [comment]
We have some Scala rules internally, but they rely on pieces that haven't yet been open sourced (not because we don't want to, but because we haven't had time - CC @IanChilds who probably knows the latest state). Until that happens, it's probably very difficult to get things working. Building on the Bazel rules might be feasible, but I don't think anyone has done that yet, so very much unknown territory.

**Reply by rockpunk** [reply]
Thanks @ndmitchell! Yeah, I've banged my head in vain against rules_scala (which is already kind of a mess as they try to port to bzlmod) and haven't made much progress. Would be happy for _any_ pointers or docs as to what scala support looks like in buck2.

I'm currently exploring a possible migration to bazel / pants / buck2 for a somewhat large codebase of python, scala, and java.
Buck2 seems to have a lot of potential and scala support or at least documentation / example would help us view buck2 more favorably in the comparison. @IanChilds, I'm happy for any info you can point me to. Thank you both!
